compat/signal.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMPAT_SIGNAL_H
+#define SQUID_COMPAT_SIGNAL_H
+
+#if !defined(SIGHUP)
```

Comment:
Please add a reference for where these values come from.

```suggestion
// Signal definitions from POSIX API. see https://www.man7.org/linux/man-pages/man7/signal.7.html
// TODO: use x86 or arch-specific defines instead of these SPARC CPU values.
#if !defined(SIGHUP)
```